### PR TITLE
Activities branch

### PR DESCRIPTION
**Pull request by Victorma**
- Sessions -> Activities
- Classes are no longer nested to games
- API changed to be consistent with the name changes and accesses.
* Classes are accesed from /classes/my
* Activities can be accesed from /activities/my, /games/:id/version/:id/activities/my and /classes/:id/activities/my
* Activities are now created by posting to /activities/ with game, version and class id as json data.
- Updated roles
- Remove Activity has been fixed.

### CODE REVIEW

File: test/tests/activitiesTest.js [hunk 1]
```diff
@@ -187,5 +238,15 @@ module.exports = function (request, db) {
...
                 });
         });
+
+        it('should DELETE an activity', function (done) {
+            request.delete('/api/activities/' + idActivity)
+                .expect(200)
+                .set('X-Gleaner-User', 'Teacher1')
+                .end(function (err, res) {
+                    should.not.exist(err);
+                    done();
```
Comment (RotaruDan) [inline test/tests/activitiesTest.js]: We should check that the activity has been deleted from mongoDB.

Also, check that the PERMISSIONS have been removed from mongo associated to that activity related with kibana visualisations/dashboards and default indices IDs.

Finally,  maybe we should also check that the data sent to that activity and "processed" by realtime to finally be written to elasticsearch has been erased (erasing the session_ID index of elasticsearch?)

File: test/tests/classesTest.js [hunk 2]
```diff
@@ -96,12 +75,13 @@ module.exports = function (request, db) {
...
         it('should GET my classes', function (done) {
-            request.get('/api/games/' + idGame + '/versions/' + idVersion + '/classes/my')
+            request.get('/api/classes/my')
                 .expect(200)
                 .set('X-Gleaner-User', 'Teacher1')
                 .end(function (err, res) {
                     should.not.exist(err);
                     should(res).be.Object();
+                    console.log(res.body);
                     should.equal(res.body.length, 1);
```
Comment (RotaruDan) [inline test/tests/classesTest.js]: Let's check the res.body[0]._id with idClass

**Comment by RotaruDan** [comment]
Review finished, minor comments pending

File: routes/classes.js [hunk 3]
```diff
@@ -30,6 +29,34 @@ var classes = require('../lib/classes');
...
+ *
+ * @apiSuccessExample Success-Response:
+ *      HTTP/1.1 200 OK
+ *      [
+ *          {
+ *              "_id": "559a447831b76cec185bf501"
+ *          },
+ *          {
+ *              "_id": "559a447831b76cec185bf511"
+ *          }
```
Comment (Victorma) [inline routes/classes.js]: Fixed in: https://github.com/e-ucm/rage-analytics-backend/pull/105/commits/7ccbeede8b109585c472b844271365b2860036ee

File: test/tests/activitiesTest.js [hunk 4]
```diff
@@ -111,8 +138,32 @@ module.exports = function (request, db) {
...
-        it('should GET my sessions', function (done) {
-            request.get('/api/games/' + idGame + '/versions/' + idVersion + '/classes/' + idClass + '/sessions/my')
+        it('should GET my activities', function (done) {
+            request.get('/api/activities/my')
+                .expect(200)
+                .set('X-Gleaner-User', 'Teacher1')
+                .end(function (err, res) {
+                    should.not.exist(err);
+                    should(res).be.Object();
+                    should.equal(res.body.length, 1);
```
Comment (Victorma) [inline test/tests/activitiesTest.js]: Fixed in: https://github.com/e-ucm/rage-analytics-backend/pull/105/commits/539d5e5e799bf6faca746fa21db733b80c11dcc4

File: test/tests/activitiesTest.js [hunk 5]
```diff
@@ -111,8 +138,32 @@ module.exports = function (request, db) {
...
+        });
+
+        it('should GET my class activities', function (done) {
+            request.get('/api/classes/' + idClass + '/activities/my')
+                .expect(200)
+                .set('X-Gleaner-User', 'Teacher1')
+                .end(function (err, res) {
+                    should.not.exist(err);
+                    should(res).be.Object();
+                    should.equal(res.body.length, 1);
```
Comment (Victorma) [inline test/tests/activitiesTest.js]: Fixed in: https://github.com/e-ucm/rage-analytics-backend/pull/105/commits/539d5e5e799bf6faca746fa21db733b80c11dcc4

File: test/tests/classesTest.js (hunk 2 above)
Comment (Victorma) [inline test/tests/classesTest.js]: Fixed in: https://github.com/e-ucm/rage-analytics-backend/pull/105/commits/57016fb261927968c5d2591adf289663f3b0be4b

File: test/tests/activitiesTest.js (hunk 1 above)
Comment (Victorma) [inline test/tests/activitiesTest.js]: How do we confirm that it has been removed? Also, I'll need more in-depth information about where and how are the PERMISSIONS stored. On the other hand, elasticsearch is really out of my knowledge so we'll have to have a private lesson. Thanks :)

File: routes/games.js [hunk 6]
```diff
@@ -66,6 +66,32 @@ router.get('/public', restUtils.find(games, function (req, callback) {
...
+ *          "title": "My Game",
+ *          "authors": ["someDeveloper"],
+ *          "developers": ["someDeveloper"],
+ *          "public": "true"
+ *      }
+ *
+ */
+router.get('/:gameId', function (req, res) {
+    var username = req.headers['x-gleaner-user'];
+    console.log(req.params.gameId + ' ' + username);
```
Comment (RotaruDan) [inline routes/games.js]: I think we should remove this console.log :)

**Pull request merged**